### DISCUSSION TITLE
What is difference between domain "expr" and domainRaw "expr"?

### DISCUSSION THREAD

**Discussion by jwoLondon** [vega/vega-lite#9055] [Q&A] [answered]
I understand that since VL 5.13 it is possible to specify a scaling domain with `domainRaw` so that it may be parameterised. But since `domain` can also take an expression as an argument, it is not clear to me what extra `domainRaw` brings. Can anyone explain the difference between the two forms, and why both are needed?

**Comment by lsh** [comment, marked as answer]
A reason to use `domainRaw` is that it doesn't overwrite the inferred domain, and when `domainRaw` evaluates to `null`, the inferred domain is used. A minimal(ish) example is the following snippet:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28},
      {"a": "B", "b": 55},
      {"a": "C", "b": 43},
      {"a": "D", "b": 91},
      {"a": "E", "b": 81},
      {"a": "F", "b": 53},
      {"a": "G", "b": 19},
      {"a": "H", "b": 87},
      {"a": "I", "b": 52}
    ]
  },
  "mark": {"type": "bar", "clip": true},
  "encoding": {
    "x": {"field": "a", "type": "nominal", "axis": {"labelAngle": 0}},
    "y": {
      "field": "b",
      "type": "quantitative",
      "scale": {"domainRaw": {"expr": "ydomain"}}
    }
  },
  "config": {
    "signals": [
      {
        "name": "ydomain",
        "update": "null",
        "on": [
          {"events": [{"type": "mousedown"}], "update": "[20, 50]"},
          {"events": [{"type": "mouseup"}], "update": "null"}
        ]
      }
    ]
  }
}
```

On `mousedown`, the chart will zoom into a predefined region, and on `mouseup` it snaps back.

https://github.com/vega/vega-lite/assets/1734032/b5e46f2e-bc8d-4daa-a014-79a3b6dbddea

With a little extra work, making custom zoom, brushing, and filtering actions becomes a little easier. To do the same thing without `domainRaw` would require getting the domain extent and storing it, and much more state management.
